finance-admin/internal/server/post_upload.go

```diff
@@ -17,7 +18,21 @@ type UploadFormHandler struct {
 func (h *UploadFormHandler) render(v AppVars, w http.ResponseWriter, r *http.Request) error {
 	ctx := r.Context()
 
+	var err error
+	var pisNumber int
 	reportUploadType := shared.ParseReportUploadType(r.PostFormValue("reportUploadType"))
+
+	if reportUploadType == shared.ReportTypeUploadPaymentsSupervisionCheque {
+		pisNumberForm := r.PostFormValue("pisNumber")
+		pisNumber, err = strconv.Atoi(pisNumberForm)
+		if len([]rune(pisNumberForm)) != 6 {
```

Comment:
Does this work if the string has whitespace? e.g. "  123 " would have a length of 6 and presumably (I haven't checked) can be parsed into a number.